spring/src/test/java/ch/qos/logback/ext/spring/MissingConfigurationTest.java

```diff
@@ -0,0 +1,179 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ch.qos.logback.ext.spring;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import ch.qos.logback.core.joran.spi.JoranException;
+import ch.qos.logback.ext.spring.web.WebLogbackConfigurer;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Random;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.impl.StaticLoggerBinder;
+import org.springframework.mock.web.MockServletContext;
+
+/**
+ * Tests the behavior of the logging configuration if the configuration files are missing.
+ * @author John Gibson
+ */
+public class MissingConfigurationTest {
+    /**
+     * Parameter specifying the location of the logback config file
+     */
+    private static final String CONFIG_LOCATION_PARAM = "logbackConfigLocation";
+
+    private static Random rng = new Random();
+
+    @ClassRule
+    public static final TemporaryFolder PLAYGROUND = new TemporaryFolder();
+
+    @Before
+    public void clearFakeLogs() {
+        FakeAppender.logs.clear();
+    }
+
+    /**
+     * Aside from printing the test name this also ensures that the logging system is in the known default state.
+     */
+    @Rule
+    public TestRule ensureLoggingInitialized = new TestWatcher() {
+        @Override
+        protected void starting(Description description) {
+            Logger log = LoggerFactory.getLogger(MissingConfigurationTest.class);
+            log.info("Starting test: " + description.getMethodName());
+        }
+    };
+
+    @After
+    public void cleanupLogging() {
+        LoggerContext loggerContext = (LoggerContext) StaticLoggerBinder.getSingleton().getLoggerFactory();
+        loggerContext.reset();
+    }
+
+    @Test(timeout = 1000L)
+    public void testMissingFileConfiguration() throws Exception {
+        MockServletContext context = new MockServletContext();
+        File missingConfig = null;
+        do {
+            missingConfig = new File("missingConfigTest" + rng.nextInt() + ".xml");
+        } while (missingConfig.exists());
+
+        context.addInitParameter(CONFIG_LOCATION_PARAM, missingConfig.toURI().toURL().toString());
+
+        WebLogbackConfigurer.initLogging(context);
+    }
+
+    @Test
+    public void testMissingClasspathConfiguration() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/tony19/logback-extensions/file/2370856094/issues/source?bid=3222649&fileBranchId=3222650#l96)